pkg/kubelet/kubelet.go

```diff
@@ -2731,7 +2741,7 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 			kl.recordNodeStatusEvent("NodeOutOfDisk")
 		}
 	} else {
-		if nodeOODCondition.Status != api.ConditionFalse {
+		if nodeOODCondition.Status != api.ConditionFalse && kl.clock.Since(nodeOODCondition.LastTransitionTime.Time) >= kl.outOfDiskTransitionFrequency {
```

Comment:
I think we should skip the clock check if nodeOODCondition.Status == api.ConditionUnknown. If we don't do that, I think on startup we will not start reporting nodeOODCondition.Status == api.ConditionFalse for five minutes, since the field is initialized to api.ConditionUnknown at startup. Same problem if the node is temporarily unreachable and NodeController marks it as nodeOODCondition.Status == api.ConditionUnknown. Do you agree?